src/librustc/middle/resolve_lifetime.rs

```diff
@@ -1861,7 +1861,7 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
             .map(|(i, input)| {
                 let mut gather = GatherLifetimes {
                     map: self.map,
-                    outer_index: ty::DebruijnIndex::INNERMOST,
+                    outer_index: ty::DebruijnIndex::innermost(),
```

Comment:
I'd prefer to keep the constant, using the macro extension I suggested in the main PR.